microbench/PR1977_bench.cpp

```
@@ -33,12 +33,12 @@ int main(int argc, char** argv) {
 	unsigned int sum = 0;
 	unsigned int TotalUsedConn = 0;
 
-	srand(monotonic_time());
+	g_seed = static_cast<unsigned int>(monotonic_time());
 	usedConns = (unsigned int *)malloc(NSRV*sizeof(unsigned int));
 	weights = (unsigned int *)malloc(NSRV*sizeof(unsigned int));
 	for (int i=0 ; i < NSRV ; i++ ) {
-		usedConns[i] = 20+rand()%1000;
-		weights[i] = 20+rand()%10000;
+		usedConns[i] = 20+fastrand()%1000;
```

Cppcheck (2.21.0)

[warning] If memory allocation fails, then there is a possible null pointer dereference

(nullPointerOutOfMemory)